src/main/resources/org/jenkinsci/plugins/badge/BadgeAction/index.groovy

```diff
@@ -30,44 +30,79 @@ l.layout {
 </style>
 """)
 
-        def base =  "${app.rootUrl}${my.project.url}";
-        def badge = base + "badge/icon"
-        
         def fullJobName = h.escape(my.project.fullName);
-        def publicbadge = "${app.rootUrl}buildStatus/icon?job=${fullJobName}";
+        def jobUrlWithView =  "${app.rootUrl}${my.project.url}";
+        def jobUrlWithoutView =  "${app.rootUrl}job/${fullJobName}";
```

Comment:
Thanks for the review. Can you provide more details where you see the problem? The code is is already used in released versions and I haven't seen any problems so far and have seen the code also in other plugin. Can you also provide more details on the workaround, i.e which code to use?